### ISSUE TITLE
_0 entries don't seem right

### ISSUE THREAD

**Issue by jfberry** [jms412/PkmnShinyMap#5]
Poracle logic is to match the entire form (ie id_form) and if that entry is not present, match just the id.
Niantic only use _0 as the form id when there are no forms, as soon as there are any forms then _0 is never used.

So if the intention is to provide form overrides then you can use the base id and the form id - but _0 as you have it will never be used (unless there are no forms in game, when it will be used as the first match)

**Comment by Fabio1988** [comment]
I agree, the actual Zorua (Dex# 570) is spawning with Form ID = 0.
Therefore the lookup without form is never successful.

So we would need an entry like 
```json
"570": " :sparkles:",
```

**Comment by jms412** [comment]
@jfberry @Fabio1988 Hey guys, I'm happy to change it if the consensus is that it is better to do so. Zorua is a good example though since we know there is a Hisuian form that will be put into the game files at some point and also released at some point. Since the Hisuian form is not in the game files yet the form ID for the normal form is still 0. Once they add the Hisuian form into the game data (which could be anytime and could even be a long time before they actually release the Hisuian form) then the normal form will become 2108. This is why I currently have both _0 and _2108 as shiny available and a place holder for the Hisuain form which contains a * so it won't trigger the shiny being available.

"570_0": " ✨",
"570_2108": " ✨",
"570_2796*": "Hisuian Form",

Once all forms have the shiny released I do update the json to just be the Pokemon ID since there is no need to check forms in that case.

Example:
"570": " ✨",

But the benefit I see to doing it this way is that Niantic rarely release Pokemon as shiny on first release. So if they do a surprise release of the non-shiny Hisuian version then the current json will process this correctly where as if I just had the Pokemon ID then it would require manual intervention to fix as the Hisuian version would be automatically showing as shiny available too.

At the end of the day I'm usually on top of any updates anyway, so probably won't hurt for me to just use the ID for the time being and then update the json once the forms are added to the game files. Let me know what you think would be best?

**Comment by Fabio1988** [comment]
Yeah Zorua is a very good example. At the moment nia introduces the hisiuan form we will never get 570_0 as result.. it will be 570_2108 and 570_2796. So it's totally safe to add 570 as key, because that is the way it is now, because hisiuan is not released yet.

But I also get how you differentiate shiny with non shiny

**Comment by jfberry** [comment]
This is a really good point from @jms412 and I agree the implementation should check _0
But it’s also safe to have the hisiuan form there already without shiny as an override with 570 without the _0 and this would be compatible with existing implementations

**Comment by jms412** [comment]
Ok so no action required by me at this stage, is that right?

**Comment by jfberry** [comment]
Correct. Poracle develop now supports _0
